tooling/nargo_cli/src/cli/test_cmd/formatters.rs

```diff
@@ -317,8 +319,132 @@ impl Formatter for TerseFormatter {
     }
 }
 
+pub(super) struct JsonFormatter;
+
+impl Formatter for JsonFormatter {
+    fn package_start(&self, package_name: &str, test_count: usize) -> std::io::Result<()> {
+        let json = json!({"type": "suite", "event": "started", "name": package_name, "test_count": test_count});
+        println!("{json}");
+        Ok(())
+    }
+
+    fn test_end(
+        &self,
+        test_result: &TestResult,
+        _current_test_count: usize,
+        _total_test_count: usize,
+        file_manager: &FileManager,
+        show_output: bool,
+        _deny_warnings: bool,
+        silence_warnings: bool,
+    ) -> std::io::Result<()> {
+        let mut json = Map::new();
+        json.insert("type".to_string(), json!("test"));
+        json.insert("name".to_string(), json!(&test_result.name));
+        json.insert("exec_time".to_string(), json!(test_result.time_to_run.as_secs_f64()));
+
+        let mut stdout = String::new();
```

Comment:
Not blocking, but I'd expect it to be useful to distinguish `stdout` from `stderr`: where `stdout` is the `test_result.output` and `stderr` includes outputs from `TestStatus::Fail` / `TestStatus::CompileError`. Otherwise, how can I get e.g. _just_ the test output from the JSON when a test fails?

Reply:
Good point. Stderr goes directly without json formatting and will only happen if compilation fails. This is the same as in Rust. So there's no need to distinguish it here (I think!)